### PR TITLE
Be able to feature test action mailbox against our example app

### PR DESCRIPTION
**Pull request by benoittgt**
Hello

We need documentation on Relish for Action Mailbox. I choose to make only one feature test file for all matchers. They are straightforward.

I had to create 3 files in my scenario, but find it the clearer to explain what we do. The result can be seen here:
https://relishapp.com/rspec-staging/rspec-rails/docs/mailbox-specs/action-mailbox-spec

Close: https://github.com/rspec/rspec-rails/issues/2341

Tasks pending:
- [x] Working feature flag

### CODE REVIEW

File: features/mailbox_specs/mailbox_spec.feature [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  `config.infer_spec_type_from_file_location!` by placing them in `spec/mailboxes`.
+
+  With mailbox specs you can:
+  * `process(mail_or_attributes)` - send mail directly to the mailbox under test for `process`ing.
+  * `receive_inbound_email(mail_or_attributes)` - matcher for asserting whether incoming email would route to the mailbox under test.
+  * `have_been_delivered` - matcher for asserting whether an incoming email object was delivered.
+  * `have_bounced` - matcher for asserting whether an incoming email object has bounced.
+  * `have_failed` - matcher for asserting whether an incoming email object has failed.
+
+  # Background:
```
Comment (benoittgt) [inline features/mailbox_specs/mailbox_spec.feature:17]: Ok, that's why the PR is in draft. I am no able to make this work, even if I was able to properly run my feature test.

Comment (JonRowe) [reply features/mailbox_specs/mailbox_spec.feature:17]: Did you define a step?

Comment (benoittgt) [reply features/mailbox_specs/mailbox_spec.feature:17]: Ok. I probably find it here
```diff
--- a/features/step_definitions/additional_cli_steps.rb
+++ b/features/step_definitions/additional_cli_steps.rb
@@ -6,6 +6,10 @@ begin
   require "action_cable"
 rescue LoadError # rubocop:disable Lint/SuppressedException
 end
+begin
+  require "action_mailbox"
+rescue LoadError # rubocop:disable Lint/SuppressedException
+end

**Comment by benoittgt** [comment]
Squashed + added to 4-0-maintenance

**Pull request merged**